### DISCUSSION TITLE
Autocomplete with limited size for huge data

### DISCUSSION THREAD

**Discussion by brunoneumann** [PhilippeMarcMeyer/vanillaSelectBox#50] [Ideas]
Hello, thanks for this nice plugin!

Can I use this plugin for a search with huge data from server? 
In this case I'd type in autocomplete input and content would be loaded on demand (showing max 10 items for example). 

I checked the ajax demo example, but it gets all data. Trying to figure it out to redraw box with new content without create a new vanillaSelectBox() instance.

**Comment by PhilippeMarcMeyer** [comment]
It would require some debouncing + ajax call with a tree redraw independant of the original SELECT TAG
(?) The already checked items would be preserved even if there are not a the search, allowing several search and check sessions
That very interesting... To study so.... By curiosity : what is you back language ?

**Reply by PhilippeMarcMeyer** [reply]
I have removed the ajax demo : I misunderstood a user question. VSB is not, at the present time, able to modify its content. The list of options comes from the original SELECT Dom element. To achieve this feature I will have to make it modify its own data AFTER init. It asks another important question : should we brake  apart from this SELECT Dom element origin and provide an alternate data input as an object ? Another point is that the result is given in the SELECT Dom element ! along with change events. With large trees the select element could be considered as a burden.

**Comment by brunoneumann** [comment]
Hello @PhilippeMarcMeyer, thanks for reply!
I thought this would be a complex change (preserve select tag + box and change only options). 
I work with meteor, I'm using typeahead autocomplete with a lot of changes to get this desired result (to look like your plugin) with large mongodb collections.
About the large trees you said, I think limiting results on server+client would be enough.

**Reply by PhilippeMarcMeyer** [reply]
the idea is to allow adding new options directly to the plugin tree but not in the original select.
But if a new option is checked, then it will be added to the select element. I you make another search maybe I should keep only the result of the search in the tree + the already checked options.
or I should consider not user the select element anymore in remote mode.

**Reply by brunoneumann** [reply]
I think if you'll change in that way, you don't need the select element anymore, but it's just thoughts because I couldn't see your entire js file yet

**Reply by PhilippeMarcMeyer** [reply]
The data remote version its on its way on the ajaxSearch branch. seems to work prety well on standard select (mulitiple and single) but it is not ready for 2 levels ones.

I wanted to start from a limited select tree coming from the select element
bu i would rather provide a data user defined fonction that allows to load a part of the tree
and then another defined user fonction for remote seraching
or maybe that would be the same ?
Still wondering

**Comment by PhilippeMarcMeyer** [comment]
I'm working on it, i wil keep you informed

**Comment by PhilippeMarcMeyer** [comment]
I'm working on another branch : ajaxSearch
I have the idea to keep on only selected options in the original select
element
on remote mode

Maybe you can take a look

Le jeu. 17 juin 2021 à 15:11, Bruno Neumann ***@***.***> a
écrit :

**Reply by PhilippeMarcMeyer** [reply]
You may test the v0.71 version if you wish : it should work form both single and multiple menus but not yet for t levels ones (optgroups)

Coming soon

**Comment by PhilippeMarcMeyer** [comment]
Hi Bruno,
Could you test the remote search
with v0.72 I think its pretty stable 
If you need any advice les me know

**Comment by PhilippeMarcMeyer** [comment]
it's nearly ready : just a few tests

**Comment by PhilippeMarcMeyer** [comment]
it's delivered ! v 0.75
You have now a remote search that works both on 1 and 2 levels menus
+ an init that can replace the origiginal select tag content (you can use an empty select tag then)
See readme for further explanations and don't hesitate to ask

**Comment by PhilippeMarcMeyer** [comment]
I won't go further in that way because I built this tool with ease of use in mind : a select element that is transformed into a dropdown and still holds the result

I may add a getChoice() function but that's all

We could build a new version with infinite levels of parents and children based on recursion
and without select element but that would be another plugin
